openmdao/utils/reports_system.py

```diff
@@ -0,0 +1,354 @@
+"""
+Utility functions related to the reporting system which generates reports by default for all runs.
+"""
+
+from collections import namedtuple, defaultdict
+from functools import wraps
+import pathlib
+import sys
+import os
+
+from openmdao.utils.mpi import MPI
+from openmdao.utils.hooks import _register_hook, _unregister_hook
+from openmdao.visualization.n2_viewer.n2_viewer import n2, _default_n2_filename
+from openmdao.visualization.scaling_viewer.scaling_report import _default_scaling_filename
+from openmdao.core.constants import _UNDEFINED
+from openmdao.core.problem import Problem
+from openmdao.core.driver import Driver
+
+# Keeping track of the registered reports
+_Report = namedtuple('Report', 'func desc class_name inst_id method pre_or_post')
+_reports_registry = {}
+
+_reports_dir = './reports'  # the default location for the reports
+
+
+def _is_rank_0(prob):
+    # Utility function to determine if MPI and on rank0 or not on MPI at all
+    on_rank0 = True
```

Comment:
Could simplify this whole function to `return not MPI or prob.comm.rank == 0`

Reply:
Great idea. Done